src/main/java/org/opensearch/knn/quantization/models/requests/TrainingRequest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.requests;
+
+import org.opensearch.knn.quantization.models.quantizationParams.QuantizationParams;
+
+/**
+ * TrainingRequest represents a request for training a quantizer.
+ *
+ * @param <T> the type of vectors to be trained.
+ */
+public abstract class TrainingRequest<T> {
+    private final QuantizationParams params;
+    private final int totalNumberOfVectors;
+    private int[] sampledIndices;
+
+    /**
+     * Constructs a TrainingRequest with the given parameters and total number of vectors.
+     *
+     * @param params              the quantization parameters.
+     * @param totalNumberOfVectors the total number of vectors.
+     */
+    protected TrainingRequest(
+            final QuantizationParams params,
+            final int totalNumberOfVectors
+    ) {
+        this.params = params;
+        this.totalNumberOfVectors = totalNumberOfVectors;
+    }
+
+    /**
+     * Returns the quantization parameters.
+     *
+     * @return the quantization parameters.
+     */
+    public QuantizationParams getParams() {
+        return params;
+    }
+
+    /**
+     * Returns the total number of vectors.
+     *
+     * @return the total number of vectors.
+     */
+    public int getTotalNumberOfVectors() {
+        return totalNumberOfVectors;
+    }
+
+    /**
+     * Sets the sampled indices for this training request.
+     *
+     * @param sampledIndices the sampled indices.
+     */
+    public void setSampledIndices(int[] sampledIndices) {
+        this.sampledIndices = sampledIndices;
+    }
```

Comment:
This setter makes the request mutable, any specific reason for this? 

Is the caller supposed to sample before calling the train of the `Quantizer`?

Reply:
Good Catch , It was just  for some testing and forgot to remove this .